<launch>
 
  <node pkg="tf" type="static_transform_publisher" name="base_to_link" args="0 0 0 0 0 0 base_footprint base_link 100" />
 	<node pkg="tf" type="static_transform_publisher" name="base_to_gyro" args="0 0 0 0 0 0 base_footprint gyro_link 100" />
  <node pkg="tf" type="static_transform_publisher" name="base_to_camera" args="0 0 0.185 0 0 0 base_footprint camera_link 100" />
  <node pkg="tf" type="static_transform_publisher" name="base_to_laser" args="0.08 0 0.14 3.1415 0 0 base_footprint laser 100" />

	<param name = "robot_description" textfile = "$(find huanyubot_description)/urdf/huanyubot.urdf"/> 
  <node name="joint_state_publisher" pkg="joint_state_publisher" type="joint_state_publisher" />
  <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher" />

  <!--Start the robot's base control code and publish the odometer data-->
  <!--node pkg="huanyu_robot_start" type="huanyu_robot_node" name="Robot_start_node" output="screen">
    <param name="usart_port"         type="string" value="/dev/huanyu_base"/>  
    <param name="baud_data"          type="int"    value="115200"/>
    <param name="robot_frame_id"     type="string" value="base_footprint"/>
    <param name="smoother_cmd_vel"   type="string" value="cmd_vel"/>
    <param name="publish_odom"       type="bool"   value="false"/>

    <param name="filter_Vx_match"    type="double" value="1.0"/>
    <param name="filter_Vy_match"    type="double" value="1.0"/>
    <param name="filter_Vth_match"   type="double" value="1.0"/>

    <param name="KP"                 type="double" value="190.0"/>
    <param name="KI"                 type="double" value="147.0"/>
    <param name="KD"                 type="double" value="54.0"/>
  </node-->

  <!-- rplidar A1-->
  <node name="rplidarNode"          pkg="rplidar_ros"  type="rplidarNode" output="screen">
    <param name="serial_port"         type="string" value="/dev/ttyUSB0"/>
    <param name="serial_baudrate"     type="int"    value="256000"/>
    <param name="frame_id"            type="string" value="laser"/>
    <param name="inverted"            type="bool"   value="false"/>
    <param name="angle_compensate"    type="bool"   value="true"/>
  </node>

  <!-- Robot pose ekf -->
  <node pkg="robot_pose_ekf" type="robot_pose_ekf" name="robot_pose_ekf" output="screen">
    <param name="freq" value="30.0"/>
    <param name="sensor_timeout" value="1.0"/>
    <param name="odom_used" value="true"/>
    <param name="imu_used" value="true"/>
    <param name="vo_used" value="false"/>
    <remap from="/imu_data" to="/mobile_base/sensors/imu_data" />
  </node>


</launch>


